cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/security/jwt/JwtAuthenticatorTest.java

```diff
@@ -0,0 +1,142 @@
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    expect(request.getHeader(HttpHeader.AUTHORIZATION.asString())).andReturn("Bearer " + EXPECTED_TOKEN);
+    replay(request);
+    String actualToken = authenticator.getJwtFromBearerAuthorization(request);
+    verify(request);
+    assertEquals(EXPECTED_TOKEN, actualToken);
+  }
+
+  @Test
+  public void testParseTokenFromAuthHeaderNoBearer() {
+    JwtAuthenticator authenticator = new JwtAuthenticator(TOKEN_PROVIDER, JWT_TOKEN);
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    expect(request.getHeader(HttpHeader.AUTHORIZATION.asString())).andReturn("Basic " + EXPECTED_TOKEN);
+    replay(request);
+    String actualToken = authenticator.getJwtFromBearerAuthorization(request);
+    verify(request);
+    assertNull(actualToken);
+  }
+
+  @Test
+  public void testParseTokenFromCookie() {
+    JwtAuthenticator authenticator = new JwtAuthenticator(TOKEN_PROVIDER, JWT_TOKEN);
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    expect(request.getCookies()).andReturn(new Cookie[] {new Cookie(JWT_TOKEN, EXPECTED_TOKEN)});
+    replay(request);
+    String actualToken = authenticator.getJwtFromCookie(request);
+    verify(request);
+    assertEquals(EXPECTED_TOKEN, actualToken);
+  }
+
+  @Test
+  public void testParseTokenFromCookieNoJwtCookie() {
+    JwtAuthenticator authenticator = new JwtAuthenticator(TOKEN_PROVIDER, JWT_TOKEN);
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    expect(request.getCookies()).andReturn(new Cookie[] {new Cookie(RANDOM_COOKIE_NAME, "")});
+    replay(request);
+    String actualToken = authenticator.getJwtFromCookie(request);
+    verify(request);
+    assertNull(actualToken);
+  }
+
+  @Test
+  public void testRedirect() throws IOException, ServerAuthException {
+    JwtAuthenticator authenticator = new JwtAuthenticator(TOKEN_PROVIDER, JWT_TOKEN);
+
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    expect(request.getMethod()).andReturn(HttpMethod.GET.asString());
+    expect(request.getQueryString()).andReturn(null);
+    expect(request.getHeader(HttpHeader.AUTHORIZATION.asString())).andReturn(null);
+    expect(request.getCookies()).andReturn(new Cookie[] {});
+    expect(request.getRequestURL()).andReturn(new StringBuffer(CRUISE_CONTROL_ENDPOINT));
+
+    HttpServletResponse response = mock(HttpServletResponse.class);
+    response.sendRedirect(TOKEN_PROVIDER.replace("{redirectUrl}", CRUISE_CONTROL_ENDPOINT));
+    expectLastCall().andVoid();
+
+    replay(request, response);
+    Authentication actualAuthentication = authenticator.validateRequest(request, response, true);
+    verify(request, response);
+    assertEquals(Authentication.SEND_CONTINUE, actualAuthentication);
+  }
+
+  @Test
+  public void testSuccessfulLogin() throws Exception {
+    UserStore testUserStore = new UserStore();
+    testUserStore.addUser(TEST_USER, null, new String[] {"USER"});
```

Comment:
(Applies to similar uses in `JwtLoginServiceTest`) Nit: Can we move hardcoded String, `"USER"`, to a static variable?